builtin/logical/ssh/util.go

```diff
@@ -0,0 +1,206 @@
+package ssh
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/vault/logical"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// Transfers the file  to the target machine by establishing an SSH
+// session with the target. Uses the public key authentication method
+// and hence the parameter 'key' takes in the private key. The fileName
+// parameter takes an absolute path.
+func uploadPublicKeyScp(publicKey, username, ip, port, key string) error {
+	dynamicPublicKeyFileName := fmt.Sprintf("vault_ssh_%s_%s.pub", username, ip)
+	session, err := createSSHPublicKeysSession(username, ip, port, key)
+	if err != nil {
+		return err
+	}
+	if session == nil {
+		return fmt.Errorf("invalid session object")
+	}
+	defer session.Close()
+	go func() {
```

Comment:
Tested it without "go func()" and it didn't work.
Since the read is from a pipe, I am guessing that io.Copy is a blocking call.
If this is the case, we are in no trouble.
Couldn't find documentation which justifies this explanation, but it is working.
What do you suggest?